src/components/comparison/comparison-side/comparison-side.tsx

```diff
@@ -393,101 +456,130 @@ export const ComparisonSide = ({
+    );
 
-            const childrenIds =
-              parentLayer?.children?.map((child) => child.id) || [];
+    if (isLayerAlreadySelected) {
+      return activeLayers.filter((activeLayer) => activeLayer.id !== layer.id);
+    }
 
-            // Find only layers which have the same current parent.
-            const childrenToSave =
-              layers.filter((layer) => childrenIds?.includes(layer.id)) || [];
+    const activeLayerIdsFromRoot = rootLayer
+      ? getAllLayersInGroupsTree(rootLayer).map((layer) => layer.id)
+      : [];
+    const activeLayersInSelectedGroup = activeLayers.filter((activeLayer) =>
+      activeLayerIdsFromRoot.includes(activeLayer.id)
+    );
 
-            const newLayers = [...childrenToSave, selectedExample];
+    return [...activeLayersInSelectedGroup, layer];
+  };
 
-            if (parentLayer) {
-              newLayers.push(parentLayer);
-            }
+  const getActiveLayerIds = (
+    layers: LayerExample[],
+    activeLayerIds: string[] = []
+  ) => {
+    for (const layer of layers) {
+      activeLayerIds.push(layer.id);
 
-            setLayers(newLayers);
-            changedLayers = newLayers;
-          }
-          break;
-        }
+      if (layer?.layers?.length) {
+        getActiveLayerIds(layer?.layers, activeLayerIds);
       }
-
-      setPreventTransitions(false);
-      onChangeLayers && changedLayers.length && onChangeLayers(changedLayers);
     }
+
+    return activeLayerIds;
   };
 
-  const getSelectedExampleById = (
-    id?: string
-  ): { selectedExample: LayerExample | null; type: LayerType } => {
-    let selectedExample: LayerExample | null = null;
-    let type: LayerType = LayerType.single;
+  const onLayerSelectHandler = (
+    layer: LayerExample,
+    rootLayer?: LayerExample
+  ) => {
+    const isGroup = !!layer?.layers?.length;
+    const isUnitLayer = !rootLayer && !isGroup;
+    const isLeaf = rootLayer && !isGroup;
+    const isMainGroup = isGroup && !rootLayer;
 
-    for (const example of examples) {
-      if (example.id === id) {
-        selectedExample = example;
+    let newActiveLayers: LayerExample[] = [];
 
-        if (example.children?.length) {
-          type = LayerType.parent;
-        }
+    switch (true) {
+      case isUnitLayer:
+        newActiveLayers = [layer];
+        break;
+      case isGroup:
+        newActiveLayers = handleSelectGroupLayer(layer, isMainGroup, rootLayer);
+        break;
+      case isLeaf:
+        newActiveLayers = handleSelectLeafLayer(layer, rootLayer);
         break;
-      }
-
-      for (const childExample of example.children || []) {
-        if (childExample.id === id) {
-          selectedExample = childExample;
-          type = LayerType.child;
-          break;
-        }
-      }
     }
 
-    return { selectedExample, type };
+    setActiveLayers(newActiveLayers);
+    setPreventTransitions(false);
+    const activeLayersIds = getActiveLayerIds(newActiveLayers);
```

Comment:
```suggestion
    const activeLayersIds = newActiveLayers.map(layer => layer.id);
```
If activeLayers are leafs of unit layers, can we just use `map` to get ids?

Reply:
done